iceoryx2-bb/container/tests/slotmap_tests.rs

```diff
@@ -0,0 +1,174 @@
+            assert_that!(sut, len i + 1);
+            assert_that!(sut.is_empty(), eq false);
+        }
+
+        assert_that!(sut.is_full(), eq true);
+        assert_that!(sut.insert(123), is_none);
+    }
+
+    #[test]
+    fn insert_when_full_fails() {
+        let mut sut = FixedSizeSut::new();
+
+        for i in 0..SUT_CAPACITY {
+            assert_that!(sut.insert(i), is_some);
+        }
+
+        assert_that!(sut.insert(34), is_none);
+    }
+
+    #[test]
+    fn removing_elements_works() {
+        let mut sut = FixedSizeSut::new();
+        let mut keys = vec![];
+
+        for i in 0..SUT_CAPACITY {
+            keys.push(sut.insert(i).unwrap());
+        }
+
+        for (n, key) in keys.iter().enumerate() {
+            assert_that!(sut.len(), eq sut.capacity() - n);
+            assert_that!(sut.is_empty(), eq false);
+            assert_that!(sut.contains(*key), eq true);
+            assert_that!(sut.remove(*key), eq true);
+            assert_that!(sut.remove(*key), eq false);
+            assert_that!(sut.contains(*key), eq false);
+            assert_that!(sut.is_full(), eq false);
+
+            assert_that!(sut.get(*key), is_none);
+            assert_that!(sut.get_mut(*key), is_none);
+        }
+
+        assert_that!(sut.is_empty(), eq true);
+    }
+
+    #[test]
+    fn removing_out_of_bounds_key_returns_false() {
+        let mut sut = FixedSizeSut::new();
+
+        assert_that!(sut.remove(SlotMapKey::new(SUT_CAPACITY + 1)), eq false);
+    }
+
+    #[test]
+    fn insert_at_works() {
+        let mut sut = FixedSizeSut::new();
+
+        let key = SlotMapKey::new(5);
+        let value = 71823;
+        assert_that!(sut.insert_at(key, 781), eq true);
+        assert_that!(sut.insert_at(key, value), eq true);
+
+        assert_that!(*sut.get(key).unwrap(), eq value);
+    }
+
+    #[test]
+    fn insert_at_and_remove_adjust_map_len_correctly() {
+        let mut sut = FixedSizeSut::new();
+
+        for n in 0..SUT_CAPACITY {
+            let key = SlotMapKey::new(n);
+            assert_that!(sut.len(), eq n);
+            assert_that!(sut.insert_at(key, 0), eq true);
+        }
+        assert_that!(sut.len(), eq SUT_CAPACITY);
+
+        for n in (0..SUT_CAPACITY).rev() {
+            let key = SlotMapKey::new(n);
+            assert_that!(sut.remove(key), eq true);
+            assert_that!(sut.remove(key), eq false);
+            assert_that!(sut.len(), eq n);
+        }
+        assert_that!(sut.len(), eq 0);
+    }
+
+    #[test]
+    fn insert_does_not_use_insert_at_indices() {
+        let mut sut = FixedSizeSut::new();
+
+        for n in 0..SUT_CAPACITY / 2 {
+            let key = SlotMapKey::new(2 * n + 1);
+            assert_that!(sut.insert_at(key, 0), eq true);
+        }
+
+        for _ in 0..SUT_CAPACITY / 2 {
+            let key = sut.insert(0);
+            assert_that!(key, is_some);
+            assert_that!(key.unwrap().value() % 2, eq 0);
+        }
+    }
```

Comment:
Please check that another insert does fail instead of using the `insert_at` keys.